src/compute-engine/boxed-expression/match.ts

```diff
@@ -299,6 +317,29 @@ function matchVariations(
   return null;
 }
 
+/**
+ *
+ * Try all needed permutations of the operands of a pattern expression, against the operands of a
+ * match target. Assumes that *expr* and *pattern* have the same operator.
+ *
+ * <!--
+ * !@fix?:
+ * - Permutations based on operands of pattern [only] do not match in some scenarios where the
+ * pattern should arguably otherwise return a valid match: specifically in cases involving patterns
+ * with seq.-wildcards. For instance:
+ *   -^A basic example is `['Add', 1, 2, 3, 'x', 5]` matched against pattern `['Add', 1, '__a']`.
+ *   Assuming the input is canonicalized, the resultant expr. against which the pattern is matched
+ *   is `['Add', 'x', 1, 2, 3, 5]`: which cannot be matched against any pattern permutation... (this
+ *   should be expected to be a valid match: with expr. expected to be commutative & therefore
+ *   operands validly taking on any arrangement, canonical-form aside).
```

Comment:
Also see opening PR message: this a remaining issue for commutative matching.

Reply:
(A bit off-topic an observation, but also noticed that for some definitions such as `Set`, these not marked as 'commutative' where they could be)